iroha-cli/interactive/interactive_transaction_cli.hpp

```diff
@@ -0,0 +1,159 @@
+
+      /**
+       * Create command menu and assign command handlers for current class
+       * object
+       */
+      void createCommandMenu();
+      /**
+       * Create result menu and assign result handlers for current class object
+       */
+      void createResultMenu();
+
+      // --- Shortcut namings for Iroha Commands ---
+      const std::string ADD_ASSET_QTY = "add_ast_qty";
+      const std::string ADD_PEER = "add_peer";
+      const std::string ADD_SIGN = "add_sign";
+      const std::string ASSIGN_M_KEY = "asn_mk";
+      const std::string CREATE_ACC = "crt_acc";
+      const std::string CREATE_ASSET = "crt_ast";
+      const std::string CREATE_DOMAIN = "crt_dmn";
+      const std::string REMOVE_SIGN = "rem_sign";
+      const std::string SET_PERM = "set_perm";
+      const std::string SET_QUO = "set_qrm";
+      const std::string SUB_ASSET_QTY = "sub_ast_qty";
+      const std::string TRAN_ASSET = "tran_ast";
+
+      // ---- Command parsers ----
+      using CommandHandler = std::shared_ptr<iroha::model::Command> (
+          InteractiveTransactionCli::*)(std::vector<std::string>);
+
+      // Specific parsers for each Iroha command
+      std::unordered_map<std::string, CommandHandler> command_handlers_;
+
+      // Descriptions for commands parameters
+      ParamsMap command_params_descriptions_;
+
+      // Descriptions for commands
+      DescriptionMap commands_description_map_;
+
+      /**
+       * Parse line with iroha Command
+       * @param line containg iroha command
+       * @return false - if parsing must be stoped, true - if parsing should
+       * continue
+       */
+      bool parseCommand(std::string line);
+      // --- Specific model Command parsers -----
+      std::shared_ptr<iroha::model::Command> parseAddAssetQuantity(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseAddPeer(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseAddSignatory(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseAssignMasterKey(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseCreateAccount(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseCreateAsset(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseCreateDomain(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseRemoveSignatory(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseSetPermissions(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseSetQuorum(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseSubtractAssetQuantity(
+          std::vector<std::string> line);
+      std::shared_ptr<iroha::model::Command> parseTransferAsset(
```

Comment:
Maybe make methods smth like: `parseStringPair` and `parseStringInt` and similar and just alias this function names to the required logic, what do you think?

Reply:
I did an attempt to do it and released there is not very convenient. From the outside (almost) all parsers are called indirectly via handleParse() template call. From the inside parsing a string or a number takes only one line and we do not have commands with a huge amount of parameters (some of them do not have parameters at all) so the resulted code will not reduce.